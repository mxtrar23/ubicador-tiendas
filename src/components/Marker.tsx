import { InfoWindowF, MarkerF } from '@react-google-maps/api'
import { MarketLocationJson } from '../vite-env'
import { useState } from 'react'
import icon from '../assets/marker.webp'

type MarkerProps = {
  data :MarketLocationJson
}
const Marker = ({data}:MarkerProps) => {
  const [showInfo, setShowInfo] = useState<boolean>(false)
  const {description,location,name} = data
  return (
    <>
      <MarkerF 
        icon={icon}
        position={location}
        title={name}
        onClick={()=>setShowInfo(true)}
        /> 
      {showInfo && <InfoWindowF
        position={location} 
        onCloseClick={()=>setShowInfo(false)}
        >
        <div className='w-full p-3 flex flex-row gap-2 text-black'>
          <div className='flex'>
            <img className='object-scale-down w-10 h-10' src={icon}/>
          </div>
          <div className='flex flex-col'>
            <h1 className='text-bold text-red-600 text-xl'>{name}</h1>
            <p >{description}</p>
          </div>
        </div>
      </InfoWindowF>}
    </>
  )
}

export default Marker