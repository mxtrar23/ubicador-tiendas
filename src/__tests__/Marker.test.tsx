import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import Marker from '../components/Marker'
import { MarketLocationJson } from '../vite-env'

vi.mock('@react-google-maps/api',()=>({
    InfoWindowF: vi.fn(({ children, onCloseClick }) => <div><button onClick={onCloseClick}>Cerrar</button>{children}</div>),
    MarkerF: vi.fn(({ title, children, onClick }) => <div><button onClick={onClick}>{title}</button>{children}</div>),
}))

describe('Marker', () => {
    it('should render', () => {

        const data: MarketLocationJson = {
            id: 1,
            description: 'test',
            location: { lat: 0, lng: 0 },
            name: 'test',
        }
        const { container } = render(<Marker data={data} />)
        expect(container).toMatchSnapshot()
    })

    it('should render Marker', async () => {
        const data: MarketLocationJson = {
            id: 1,
            description: 'mi descripcion',
            location: { lat: 0, lng: 0 },
            name: 'mi titulo',
        }
        render(<Marker data={data} />)

        const openBotton = screen.getByText(data.name)

        fireEvent(openBotton, new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
        }))

        await waitFor(() => {
            screen.getByText(data.description)
        })

        const closeBotton = screen.getByText('Cerrar')

        fireEvent(closeBotton, new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
        }))


        waitFor(() => {
            expect(screen.getAllByText(data.description)).toBeNull();
        })

    })


})