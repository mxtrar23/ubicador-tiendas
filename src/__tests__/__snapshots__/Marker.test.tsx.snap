// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Marker > should render 1`] = `
<div>
  <div>
    <button>
      test
    </button>
  </div>
</div>
`;
